Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -546,15 +546,18 @@ namespace AzToolsFramework
                 templateId, templateToDelete.GetFilePath().c_str());
 
             m_templateInstanceMapper.UnregisterTemplate(templateId);
-            
+
             result = m_templateIdMap.erase(templateId) != 0;
             AZ_Assert(result,
                 "Prefab - PrefabSystemComponent::RemoveTemplate - "
                 "Failed to remove Template with Id '%llu' on file path '%s' "
                 "from Template Id Map.",
                 templateId, templateToDelete.GetFilePath().c_str());
 
-            return;
+            if (!m_removingAllTemplates)
```

Comment:
This pattern can be confusing for the listeners of this ebus. If all templates are removed, then we should still send notification per template removal. There can be listeners who just care about a single template. They will never receive the 'OnTemplateRemoved' notification if that template was removed in a batch. Therefore, the 'm_removingAllTemplates' boolean should be removed altogether imo.

Reply:
Having both fire kind of defeats the purpose of having the OnAllTemplatesRemoved since by the time you get that event, you've already handled each individually and now can't batch handle them, since you have no idea that event is coming beforehand.  I would argue someone shouldn't be listening to only OnTemplateRemoved, just as they shouldn't listen to only OnAllTemplatesRemoved.